services/api/db/Oauth/register.py

```diff
@@ -0,0 +1,51 @@
+from django.contrib.auth import authenticate
+from db.models import User
+import random
+from rest_framework.exceptions import AuthenticationFailed
+from kalvi.api.views.auth import get_tokens_for_user
+
+
+def generate_username(name):
+    name = "".join(name.split(' ')).lower()
+    if not User.objects.filter(name=name).exists():
+        return name
+    else:
+        random_name = name + str(random.randint(0, 1000))
+        return generate_username(random_name)
+
+def register_social_user(provider, email, name):
+    filtered_user_by_email = User.objects.filter(email=email)
+    if filtered_user_by_email.exists():
+        if provider == filtered_user_by_email[0].auth_provider:
+            registered_user = authenticate(email=email, password="SECRET")
```

Comment:
```suggestion
            registered_user = authenticate(email=email, password="SECRET")
```
should the password string be obtained from env rather than defining it here?